### PR TITLE
[labs/analyzer] Fix resolution and module detection w/ NodeNext resolution

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes https://github.com/lit/lit/issues/4742
Fixes https://github.com/lit/lit/issues/4743

### CODE REVIEW

File: packages/labs/analyzer/test-files/ts/node-next-resolution/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import '../element-a.js';
+
+// This must infer as ElementA i norder to trigger
```
Comment (aomarks) [inline packages/labs/analyzer/test-files/ts/node-next-resolution/src/main.ts:9]:
```suggestion
// This must infer as ElementA in order to trigger
```

File: packages/labs/analyzer/test-files/ts/node-next-resolution/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
```
Comment (aomarks) [inline packages/labs/analyzer/test-files/ts/node-next-resolution/src/main.ts:3]:
```suggestion
 * Copyright 2024 Google LLC
```

File: packages/labs/analyzer/test-files/ts/node-next-resolution/src/element-a.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
```
Comment (aomarks) [inline packages/labs/analyzer/test-files/ts/node-next-resolution/src/element-a.ts:3]:
```suggestion
 * Copyright 2024 Google LLC
```

**Review by aomarks — APPROVED** [review]

File: packages/labs/analyzer/src/lib/types.ts [hunk 4]
```diff
@@ -199,7 +199,19 @@ const getSpecifierFromTypeImport = (
...
+    // Types can resolve to source or declaration files, so include those
+    // extensions as well. We still include JS extensions to support JSDoc
+    // types.
+    // We don't include the CommonJS-only extensions .cts and .d.cts here
+    resolveExtension(specifier as AbsolutePath, analyzer, [
+      'js',
+      'mjs',
+      'ts',
+      'mts',
+      'tsx',
```
Comment (kevinpschaaf) [inline packages/labs/analyzer/src/lib/types.ts:211]: Need jsx too then?
```suggestion
      'tsx',
      'jsx',
```

**Review by kevinpschaaf — APPROVED** [review]
LGTM!

**Pull request merged**